models/repo/archive_download_count.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/git"
+	api "code.gitea.io/gitea/modules/structs"
+)
+
+// RepoArchiveDownloadCount counts all archive downloads for a tag
+type RepoArchiveDownloadCount struct { //nolint:revive
+	ID     int64           `xorm:"pk autoincr"`
+	RepoID int64           `xorm:"index unique(s)"`
+	Type   git.ArchiveType `xorm:"unique(s)"`
+	Tag    string          `xorm:"index unique(s)"`
```

Comment:
Tag or commit?

Reply:
Tag. I think it would not make much sense to count the downloads for each commit.

Reply:
Why not use a release_id ?

Reply:
That would mean that we need to query to database for the Release ID at some positions, including convert.

Reply:
Ping @lunny

Reply:
> That would mean that we need to query to database for the Release ID at some positions, including convert.

Yes, it will be faster to store an int64 id?